Game/Source/GDKTestGyms/BenchmarkGymGameModeBase.cpp

```diff
@@ -488,22 +488,27 @@ void ABenchmarkGymGameModeBase::TickUXMetricCheck(float DeltaSeconds)
 
 void ABenchmarkGymGameModeBase::TickActorCountCheck(float DeltaSeconds)
 {
-	const UNFRConstants* Constants = UNFRConstants::Get(GetWorld());
-	check(Constants);
-	FMetricTimer TempActorCheckDelay = Constants->ActorCheckDelay;
 	// This test respects the initial delay timer in both native and GDK
-	if (TempActorCheckDelay.HasTimerGoneOff() &&
-		!TestLifetimeTimer.HasTimerGoneOff())
+	if (TickActorCountTimer.HasTimerGoneOff())
 	{
 		TryInitialiseExpectedActorCounts();
-		
-		const UWorld* World = GetWorld();
-		const FString WorkerID = GetGameInstance()->GetSpatialWorkerId();
-		const int32 ActualCountSimulate = GetActorClassCount(ExpectedSimPlayersCount.ActorClass);
-		const int32 ActualCountNPCs = GetActorClassCount(ExpectedNPCsCount.ActorClass);
-		GenerateTotalNumsForActors(WorkerID, World, ExpectedSimPlayersCount, MapAuthoritativeSimPlayers, SmoothedTotalAuthSimPlayers, ActualCountSimulate, false);
-		ReportAuthoritativeNPCs(WorkerID, World, ActualCountNPCs);
-		TempActorCheckDelay.SetTimer(1);
+
+		if (HasAuthority())
+		{
+			ActorCountReportIdx++;
+			UpdateAndReportActorCounts();
+
+			const int32 TickActorCountPeriodInSeconds = 10;
+			TickActorCountTimer.SetTimer(TickActorCountPeriodInSeconds);
+
+			TimeSinceLastCheckedTotalActorCounts += DeltaSeconds;
```

Comment:
This will only increment on the frames that `TickActorCountTimer.HasTimerGoneOff()` is true, which is only every 10 seconds right? So can't use DeltaSeconds here, but need to use absolute game time. I'd recommend just switching this to a timer on the timermanager, that you reset in `ReportAuthoritativeActorCount`. Here's an example of how we used to use this same pattern in the heartbeat manager - https://github.com/spatialos/UnrealGDK/blob/925bba1ab2ab4ba1ed33758624478910ef5e2ecf/SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetConnection.cpp#L148